src/bin/inx-chronicle/cli.rs

```diff
@@ -281,58 +292,94 @@ impl ClArgs {
 
                     let num_tasks = num_tasks.unwrap_or(1);
                     let mut join_set = tokio::task::JoinSet::new();
+                    let chunk_size = (end_milestone.0 - start_milestone.0) / num_tasks as u32
+                        + ((end_milestone.0 - start_milestone.0) % num_tasks as u32 != 0) as u32;
                     for i in 0..num_tasks {
                         let db = db.clone();
                         let influx_db = influx_db.clone();
-                        let analytics_choice = analytics.clone();
+                        let mut analytics_choices = if analytics.is_empty() {
+                            influxdb::all_analytics()
+                        } else {
+                            analytics.iter().copied().collect()
+                        };
+
                         join_set.spawn(async move {
-                            let mut analytics = if analytics_choice.is_empty() {
-                                chronicle::db::collections::analytics::all_analytics()
-                            } else {
-                                let mut tmp: std::collections::HashSet<
-                                    chronicle::db::influxdb::config::AnalyticsChoice,
-                                > = analytics_choice.iter().copied().collect();
-                                tmp.drain().map(Into::into).collect()
-                            };
+                            tracing::info!("Computing the following analytics: {:?}", analytics_choices);
 
-                            tracing::info!("Computing the following analytics: {:?}", analytics);
+                            let start_milestone = start_milestone + i as u32 * chunk_size;
 
-                            for index in (*start_milestone..*end_milestone).skip(i).step_by(num_tasks) {
-                                let milestone_index = index.into();
-                                if let Some(milestone_timestamp) = db
-                                    .collection::<chronicle::db::collections::MilestoneCollection>()
-                                    .get_milestone_timestamp(milestone_index)
-                                    .await?
-                                {
-                                    #[cfg(feature = "metrics")]
-                                    let start_time = std::time::Instant::now();
+                            let mut state: Option<crate::stardust_inx::AnalyticsState> = None;
+
+                            let tangle = Tangle::from_mongodb(&db);
+                            let mut milestone_stream = tangle
+                                .milestone_stream(start_milestone..start_milestone + chunk_size)
+                                .await?;
+                            use futures::TryStreamExt;
```

Comment:
Let's move the `fill-analytics` logic to a separate file then we also don't have to do these `use` shenanigans.